modules/auxiliary/gather/python_flask_cookie_signer.rb

```diff
@@ -45,54 +45,62 @@ def initialize(info = {})
       [
         Opt::RPORT(80),
         OptString.new('TARGETURI', [ true, 'URI to browse', '/']),
-        OptString.new('NEWCOOKIECONTENT', [ false, 'Content of a cookie to sign', '']),
-        OptString.new('SECRET', [ false, 'Content of a cookie to sign', '']),
+        OptString.new('NEWCOOKIECONTENT', [ false, 'Content of a cookie to sign', ''], conditions: %w[ACTION == Resign]),
+        OptString.new('SECRET', [ true, 'The key with which to sign the cookie', '']),
+      ]
+    )
+    register_advanced_options(
+      [
+        OptString.new('CookieName', [ true, 'The name of the session cookie', 'session' ]),
+        OptString.new('Salt', [ true, 'The salt to use for key derivation', Msf::Exploit::Remote::HTTP::FlaskUnsign::Session::DEFAULT_SALT ])
       ]
     )
   end
 
-  def check
-    res = send_request_cgi!({
-      'uri' => normalize_uri(target_uri.path)
-    })
-    return Exploit::CheckCode::Unknown("#{peer} - Could not connect to web service - no response") if res.nil?
-    return Exploit::CheckCode::Unknown("#{peer} - Unexpected response code (#{res.code})") unless res.code == 200
-    return Exploit::CheckCode::Safe("#{peer} - Unexpected response, version_string not detected") unless res.body.include? 'version_string'
-    unless res.body =~ /&#34;version_string&#34;: &#34;([\d.]+)&#34;/
-      return Exploit::CheckCode::Safe("#{peer} - Unexpected response, unable to determine version_string")
-    end
-
-    version = Rex::Version.new(Regexp.last_match(1))
-    if version < Rex::Version.new('2.0.1') && version >= Rex::Version.new('1.4.1')
-      Exploit::CheckCode::Vulnerable("Apache Supset #{version} is vulnerable")
-    else
-      Exploit::CheckCode::Safe("Apache Supset #{version} is NOT vulnerable")
-    end
-  end
-
-  def retrieve
+  def action_retrieve
     print_status("#{peer} - Retrieving Cookie")
     res = send_request_cgi!({
       'uri' => normalize_uri(target_uri.path),
       'keep_cookies' => true
     })
     fail_with(Failure::Unreachable, "#{peer} - Could not connect to web service - no response") if res.nil?
     fail_with(Failure::UnexpectedReply, "#{peer} - Unexpected response code (#{res.code})") unless res.code == 200
-    cookie = res.get_cookies.to_s
+
+    cookie = cookie_jar.cookies.find { |c| c.name == datastore['CookieName'] }&.cookie_value
+    fail_with(Failure::UnexpectedReply, "#{peer} - Response is missing the session cookie") unless cookie
```

Comment:
This was necessary to handle instances where the server returned more than one cookie.